hugegraph-core/src/main/java/org/apache/hugegraph/traversal/algorithm/EdgeExistenceTraverser.java

```diff
@@ -0,0 +1,49 @@
+package org.apache.hugegraph.traversal.algorithm;
+
+import org.apache.hugegraph.HugeGraph;
+import org.apache.hugegraph.backend.id.Id;
+import org.apache.hugegraph.backend.query.ConditionQuery;
+import org.apache.hugegraph.backend.query.QueryResults;
+import org.apache.hugegraph.iterator.ListIterator;
+import org.apache.hugegraph.schema.EdgeLabel;
+import org.apache.hugegraph.type.HugeType;
+import org.apache.hugegraph.type.define.Directions;
+import org.apache.hugegraph.type.define.HugeKeys;
+import org.apache.tinkerpop.gremlin.structure.Edge;
+
+import java.util.Iterator;
+import java.util.List;
+
+public class EdgeExistenceTraverser extends HugeTraverser {
+    public EdgeExistenceTraverser(HugeGraph graph) {
+        super(graph);
+    }
+
+    public Iterator<Edge> queryEdgeExistence(Id sourceId, Id targetId, String label,
+                                             String sortValues, long limit) {
+        if (label == null || label.isEmpty()) {
+            return queryByNeighbor(sourceId, targetId, limit);
+        }
+        Id edgeLabelId = getEdgeLabelId(label);
+        EdgeLabel edgeLabel = EdgeLabel.undefined(graph(), edgeLabelId);
+        ConditionQuery conditionQuery = new ConditionQuery(HugeType.EDGE);
+        conditionQuery.eq(HugeKeys.OWNER_VERTEX, sourceId);
+        conditionQuery.eq(HugeKeys.OTHER_VERTEX, targetId);
+        conditionQuery.eq(HugeKeys.LABEL, edgeLabelId);
+        conditionQuery.eq(HugeKeys.DIRECTION, Directions.OUT);
+        conditionQuery.limit(limit);
+        if (edgeLabel.existSortKeys()) {
+            conditionQuery.eq(HugeKeys.SORT_VALUES, sortValues);
+        }else {
+            conditionQuery.eq(HugeKeys.SORT_VALUES, "");
+        }
+        return graph().edges(conditionQuery);
+    }
+
+    private Iterator<Edge> queryByNeighbor(Id sourceId, Id targetId, long limit) {
+        ListIterator<Edge> edges = QueryResults.toList(this.edgesOfVertex(sourceId, Directions.OUT, (Id) null, limit));
+        List<Edge> res = newList();
+        edges.forEachRemaining(edge -> {if (targetId.compareTo((Id) edge.inVertex().id()) == 0)  res.add(edge);});
```

Comment:
can we use for-loop instead? to keep a consistent style and reduce the times of callbacks